lib/facebook_canvas/middleware.rb

```diff
@@ -12,24 +12,58 @@
 # Addionally you can restrict `REQUEST_METHOD` rewriting by providing a `custom_filter` block.
 module FacebookCanvas
   class Middleware
-    def initialize(app, request_host, custom_filter)
+    ENV_INSIDE = "facebook_canvas.inside".freeze
+
+    def initialize(app, request_host, custom_filter, inside_filter: nil)
       @app = app
-      @request_host = request_host
+      @request_host = proc_for_request_host(request_host)
       @custom_filter = custom_filter
+      @inside_filter = inside_filter
     end
 
     # Forces REQUEST_METHOD to GET if required.
     def call(env)
-      if matches_server_name?(env) && was_get_request?(env) && !was_xhr_request?(env) && custom_filter?(env)
+      inside = match_inside?(env)
```

Comment:
Yes, that's true.

As the variable `inside` is used later, I'd like to increase the visibility of this variable by putting it in its own line.

Thoughts?